### PR TITLE
Tree widget: focus grouping node

### PR DESCRIPTION
**Pull request by saskliutas**
Closes https://github.com/iTwin/viewer-components-react/issues/884

Additionaly added Icon for grouping nodes, ran prettier.

### CODE REVIEW

File: packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTreeDefinition.ts [hunk 1]
```diff
@@ -598,9 +604,20 @@ function createGeometricElementInstanceKeyPaths(
...
 ): Observable<HierarchyNodeIdentifiersPath> {
   return defer(() => {
+    const elementIds = elementInfos.filter((info): info is Id64String => typeof info === "string");
+    const groupInfos = elementInfos.filter((info): info is ElementsGroupInfo => typeof info !== "string");
+
+    const elementsClause = elementIds.length > 0 ? `e.ECInstanceId IN (${elementIds.map(() => "?").join(",")})` : undefined;
+    const classClause =
+      groupInfos.length > 0
+        ? groupInfos.map((info) => `(${info.parentType === "element" ? "e.Parent.Id" : "e.Category.Id"} = ? AND e.ECClassId IS (${info.classes.join(",")}))`)
+        : undefined;
```
Comment (grigasp) [inline packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTreeDefinition.ts:617]:
```suggestion
    const classClause = groupInfos.map((info) => `(${info.parentType === "element" ? "e.Parent.Id" : "e.Category.Id"} = ? AND e.ECClassId IS (${info.classes.join(",")}))`);
```

File: packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTree.tsx [hunk 2]
```diff
@@ -210,3 +216,42 @@ function useCachedVisibility(activeView: Viewport, hierarchyConfig: ModelsTreeHi
...
+    if (!HierarchyNodeKey.isClassGrouping(key.key)) {
+      targetKeys.push(...key.groupedInstanceKeys);
+      continue;
+    }
+
+    if (!key.nonGroupingAncestor || !HierarchyNodeKey.isInstances(key.nonGroupingAncestor.key)) {
+      continue;
+    }
+
+    const parentKey = key.nonGroupingAncestor.key.instanceKeys[0];
```
Comment (grigasp) [inline packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTree.tsx:247]: Category nodes may be merged from multiple category instances - can we handle that here?

File: packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTree.tsx [hunk 3]
```diff
@@ -210,3 +216,42 @@ function useCachedVisibility(activeView: Viewport, hierarchyConfig: ModelsTreeHi
...
+    const type = key.nonGroupingAncestor.extendedData?.isCategory ? "category" : "element";
+    const targetInfo = targetKeys.find((target): target is ElementsGroupInfo => !("id" in target) && target.parentKey.id === parentKey.id);
+    if (targetInfo) {
+      targetInfo.classes.push(key.key.className);
```
Comment (grigasp) [inline packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTree.tsx:251]: shouldn't we worry about duplicate class names here?

File: packages/itwin/tree-widget/src/test/trees/models-tree/stateless/ModelsTreeFiltering.test.ts [hunk 4]
```diff
@@ -910,14 +967,290 @@ describe("Models tree", () => {
...
+                        label: "Physical Object",
+                        autoExpand: true,
+                        children: [
+                          NodeValidators.createForInstanceNode({
+                            instanceKeys: [x.rootElement],
+                            label: /^root element/,
+                            children: [
+                              NodeValidators.createForClassGroupingNode({
+                                label: "Test Physical Object",
+                                autoExpand: true,
```
Comment (grigasp) [inline packages/itwin/tree-widget/src/test/trees/models-tree/stateless/ModelsTreeFiltering.test.ts:1112]: Ideally, the hierarchy should be auto-expanded up to the grouping node, but the grouping node itself should not. Can we do that?

File: packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTreeDefinition.ts (hunk 1 above)
Comment (saskliutas) [reply packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTreeDefinition.ts]: Changed

File: packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTree.tsx (hunk 2 above)
Comment (saskliutas) [reply packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTree.tsx]: Updated to handle multiple instances

File: packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTree.tsx (hunk 3 above)
Comment (saskliutas) [reply packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTree.tsx]: I assume there should not be multiple grouping nodes for same class under same non-grouping parent

File: packages/itwin/tree-widget/src/test/trees/models-tree/stateless/ModelsTreeFiltering.test.ts (hunk 4 above)
Comment (grigasp) [reply packages/itwin/tree-widget/src/test/trees/models-tree/stateless/ModelsTreeFiltering.test.ts:1178]: Filed a separate issue for tracking this: https://github.com/iTwin/viewer-components-react/issues/947

**Review by grigasp — APPROVED** [review]

**Pull request merged**